### ISSUE TITLE
Settings page is blank

### ISSUE THREAD

**Issue by JackSW** [deliciousbrains/wp-amazon-s3-and-cloudfront#218]
Settings page does not render and the log files show the following error:

PHP Catchable fatal error:  Object of class WP_Error could not be converted to string in /wordpress/wp-content/plugins/amazon-web-services/vendor/aws/Aws/Common/RulesEndpointProvider.php on line 65, referer: http://website.com/wp-admin/admin.php?page=amazon-web-services

I tracked down that the error gets trigger in amazon-s3-and-cloudfront/view/settings.php when calling $this->check_write_permission();

WP: v4.1
Amazon Web Services: v0.2.1
Amazon S3 and CloudFront: v0.8

![1-13-2015 5-54-35 pm](https://cloud.githubusercontent.com/assets/10355266/5732792/48ed5f96-9b4d-11e4-9c60-c575f362af58.png)

**Comment by ianmjones** [comment]
Hey @JackSW, what web server are you using?

**Comment by JackSW** [comment]
Apache/2.2.22 (Ubuntu) on EC2

**Comment by kersten** [comment]
Same problem here

**Comment by kersten** [comment]
I could get this fixed if I give all permissions of S3 of all buckets to the user I'm using.

My policy now look like this:

```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "Stmt1409910033000",
      "Effect": "Allow",
      "Action": [
        "s3:*"
      ],
      "Resource": [
        "arn:aws:s3:::*"
      ]
    }
  ]
}
```

This is not OK for me  I want to use something like this:

```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "Stmt1409910033000",
      "Effect": "Allow",
      "Action": [
        "s3:*"
      ],
      "Resource": [
        "arn:aws:s3:::my-customers-bucket/*"
      ]
    },{
      "Sid": "Stmt1409910304000",
      "Effect": "Allow",
      "Action": [
        "s3:ListAllMyBuckets"
      ],
      "Resource": [
        "arn:aws:s3:::*"
      ]
    }
  ]
}
`````
Could you please fix this issue?

**Comment by polevaultweb** [comment]
@kersten are you hosting the site on EC2 aswell?

**Comment by kersten** [comment]
Yes, Auto deployed by opsworks

**Comment by polevaultweb** [comment]
@kersten thanks! Before you fixed the issue with the policy, did you get the same error in your log?
> PHP Catchable fatal error: Object of class WP_Error could not be converted to string in /wordpress/wp-content/plugins/amazon-web-services/vendor/aws/Aws/Common/RulesEndpointProvider.php on line 65, referer: http://edit.viewmixed.com/wp-admin/admin.php?page=amazon-web-services

**Comment by polevaultweb** [comment]
@kersten just to confirm, this policy does not work for you?
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "Stmt1409910033000",
      "Effect": "Allow",
      "Action": [
        "s3:*"
      ],
      "Resource": [
        "arn:aws:s3:::my-customers-bucket/*"
      ]
    },{
      "Sid": "Stmt1409910304000",
      "Effect": "Allow",
      "Action": [
        "s3:ListAllMyBuckets"
      ],
      "Resource": [
        "arn:aws:s3:::*"
      ]
    }
  ]
}
```

Can you upload me a screenshot of the settings page when you are using that policy and let me know what the region of your bucket is, and the region of your EC2 instance please?

**Comment by kersten** [comment]
Correct. This policy does not work. I don't know if the code checks different permissions, but it should work with this type of policy.

And I don't get the error ...

**Comment by polevaultweb** [comment]
@kersten what are you seeing then, just a blank settings page with that policy?

**Comment by kersten** [comment]
yeah, just a white page. No PHP Error or something like that.

**Comment by polevaultweb** [comment]
@kersten cheers. What is the region of your chosen bucket, and the region of your EC2 instance so I can try to recreate please

**Comment by kersten** [comment]
@polevaultweb both are in Ireland. Here are some facts about the machine:

######  OpsWorks Summary  ######
   Operating System: Ubuntu 14.04.1 LTS
   ......
   EC2 Region: eu-west-1
   EC2 Availability Zone: eu-west-1a

**Comment by polevaultweb** [comment]
@kersten cheers!

**Comment by polevaultweb** [comment]
@kersten would you mind testing with these versions for the plugin to see if they fix your issue?
http://cl.ly/3b1W0q2d2A1W
http://cl.ly/3g1Q303M1215

**Comment by kersten** [comment]
sorry @polevaultweb. Already fixed? Got two updates. Do I need to check this stuff you've uploaded here?

**Comment by polevaultweb** [comment]
@kersten the releases yesterday should fix the issues, update and let me know :)

**Comment by A5hleyRich** [comment]
Fixed in version 0.8.1.
